File: ios/SharedModule.m
//
//  SharedModule.m
//  FakeHunter
//
//  Created by Kamil Cichuta on 20/07/2020.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(SharedModule,NSObject)
RCT_EXTERN_METHOD(getShareUrl: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(clearShareUrl)
RCT_EXTERN_METHOD(getOpenUrl: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(clearOpenUrl)
@end

File: ios/SharedModule.swift
//
//  SharedModule.swift
//  FakeHunter
//
//  Created by Kamil Cichuta on 20/07/2020.
//

import Foundation
@objc(SharedModule) class SharedModule: NSObject {
  
  let groupId = "group.fakehunter.share"
  let shareKey = "shareUrl"
  let openKey = "openUrl"
  
  @objc func getShareUrl(_ callback: RCTResponseSenderBlock) {
    if let sharedDefaults = UserDefaults(suiteName: groupId) {
      let url = sharedDefaults.string(forKey: shareKey)
      if (url != nil) {
        return callback([NSNull(), url])
      }
    }
  }
  
  @objc func clearShareUrl() {
    if let sharedDefaults = UserDefaults(suiteName: groupId) {
      sharedDefaults.set(nil, forKey: shareKey)
      sharedDefaults.synchronize()
    }
  }
  
  @objc func getOpenUrl(_ callback: RCTResponseSenderBlock) {
    let sharedDefaults = UserDefaults.standard
    let url = sharedDefaults.string(forKey: openKey)
    if (url != nil) {
      return callback([NSNull(), url])
    }
  }
  
  @objc func clearOpenUrl() {
    let sharedDefaults = UserDefaults.standard
    sharedDefaults.set(nil, forKey: openKey)
    sharedDefaults.synchronize()
  }
  
}
